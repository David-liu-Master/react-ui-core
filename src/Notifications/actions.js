import {
  ADD_NOTIFICATION,
  REMOVE_NOTIFICATION,
  CLEAR_NOTIFICATIONS
} from './constants';

export function addNotification(notification) {
  return {
    type: ADD_NOTIFICATION,
    payload: {
      notification
    }
  };
}

export function removeNotification() {
  return {
    type: REMOVE_NOTIFICATION
  };
}

export function clearNotifications() {
  return {
    type: CLEAR_NOTIFICATIONS
  };
}
