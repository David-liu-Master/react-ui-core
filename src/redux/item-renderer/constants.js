export const CHANGE_NOTE = 'openpatch.ui-core.CHANGE_NOTE';
export const CHANGE_RESPONSE = 'openpatch.ui-core.CHANGE_RESPONSE';
