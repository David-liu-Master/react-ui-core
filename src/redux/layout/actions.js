import {
  TOGGLE_DRAWER,
  REMOTE_SECTION_REQUEST,
  REMOTE_SECTION_ROLLBACK,
  REMOTE_SECTION_COMMIT
} from './constants';

export function toggleDrawer() {
  return {
    type: TOGGLE_DRAWER
  };
}

export function fetchRemoteSection(url) {
  return dispatch => {
    dispatch(fetchRemoteSectionRequest());
    fetch(url, {
      method: 'GET',
      cors: 'no-cors'
    })
      .then(function(response) {
        return response.json();
      })
      .then(function(json) {
        return dispatch(fetchRemoteSectionCommit(url, json));
      })
      .catch(function(err) {
        return dispatch(fetchRemoteSectionRollback());
      });
  };
}

function fetchRemoteSectionRequest() {
  return {
    type: REMOTE_SECTION_REQUEST
  };
}

function fetchRemoteSectionCommit(url, sections) {
  return {
    type: REMOTE_SECTION_COMMIT,
    payload: {
      url,
      sections
    }
  };
}

function fetchRemoteSectionRollback() {
  return {
    type: REMOTE_SECTION_ROLLBACK
  };
}
