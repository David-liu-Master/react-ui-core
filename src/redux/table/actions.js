import {
  INIT,
  SELECT_ROWS,
  DESELECT_ROWS,
  SET_PAGE,
  SET_PAGE_SIZE,
  SET_ROWS_COUNT,
  SORT,
  ROWS_REQUEST,
  ROWS_COMMIT,
  ROWS_ROLLBACK,
  REFRESH
} from './constants';

export function init(tableId) {
  return {
    type: INIT,
    payload: {
      tableId
    }
  };
}

export function selectRows(tableId, rowIds) {
  return {
    type: SELECT_ROWS,
    payload: {
      tableId,
      rowIds
    }
  };
}

export function deselectRows(tableId, rowIds) {
  return {
    type: DESELECT_ROWS,
    payload: {
      tableId,
      rowIds
    }
  };
}

export function setPage(tableId, page) {
  return {
    type: SET_PAGE,
    payload: {
      tableId,
      page
    }
  };
}

export function setPageSize(tableId, pageSize) {
  return {
    type: SET_PAGE_SIZE,
    payload: {
      tableId,
      pageSize
    }
  };
}

export function setRowsCount(tableId, rowsCount) {
  return {
    type: SET_ROWS_COUNT,
    payload: {
      tableId,
      rowsCount
    }
  };
}

export function sort(tableId, order, orderBy) {
  return {
    type: SORT,
    payload: {
      tableId,
      order,
      orderBy
    }
  };
}

export function getRows(tableId, axios, url, rowsSelector, countSelector) {
  return dispatch => {
    dispatch(getRowsRequest(tableId));
    return axios
      .get(url)
      .then(response => {
        dispatch(setRowsCount(tableId, countSelector(response.data)));
        return dispatch(getRowsCommit(tableId, rowsSelector(response.data)));
      })
      .catch(() => dispatch(getRowsRollback(tableId)));
  };
}

function getRowsRequest(tableId) {
  return {
    type: ROWS_REQUEST,
    payload: {
      tableId
    }
  };
}

function getRowsCommit(tableId, rows) {
  return {
    type: ROWS_COMMIT,
    payload: {
      tableId,
      rows
    }
  };
}

function getRowsRollback(tableId) {
  return {
    type: ROWS_ROLLBACK,
    payload: {
      tableId
    }
  };
}

export function refresh(tableId) {
  return {
    type: REFRESH,
    payload: {
      tableId
    }
  };
}
