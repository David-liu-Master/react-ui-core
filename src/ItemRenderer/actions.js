import { CHANGE_NOTE, CHANGE_RESPONSE } from './constants';

export const changeNote = note => ({
  type: CHANGE_NOTE,
  payload: {
    note
  }
});

export const changeResponse = (response, id) => ({
  type: CHANGE_RESPONSE,
  payload: {
    response,
    id
  }
});
