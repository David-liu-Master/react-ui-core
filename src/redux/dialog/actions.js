import { INIT, OPEN, CANCEL, CONFIRM } from './constants';

export function init(dialogId) {
  return {
    type: INIT,
    payload: {
      dialogId
    }
  };
}

export function open(dialogId) {
  return {
    type: OPEN,
    payload: {
      dialogId
    }
  };
}

export function confirm(dialogId) {
  return {
    type: CONFIRM,
    payload: {
      dialogId
    }
  };
}

export function cancel(dialogId) {
  return {
    type: CANCEL,
    payload: {
      dialogId
    }
  };
}
