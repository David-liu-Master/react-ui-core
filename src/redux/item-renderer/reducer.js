import produce from 'immer';

import { CHANGE_NOTE, CHANGE_RESPONSE } from './constants';

const initalState = { responses: {}, note: '' };

export default (state = initalState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case CHANGE_NOTE:
        draft.note = action.payload.note;
        break;
      case CHANGE_RESPONSE:
        draft.responses[action.payload.id] = action.payload.response;
        break;
    }
  });
