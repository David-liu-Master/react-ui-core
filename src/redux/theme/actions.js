import { CHANGE_THEME, RESET_THEME } from './constants';

export function changeTheme(primary, secondary) {
  return {
    type: CHANGE_THEME,
    payload: {
      primary,
      secondary
    }
  };
}

export function resetTheme() {
  return {
    type: RESET_THEME
  };
}
