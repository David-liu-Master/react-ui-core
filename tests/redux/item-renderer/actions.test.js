import { changeNote, changeResponse } from '../../../src/redux/item-renderer/actions';

describe('ItemRenderer actions', () => {
  it('should create changeNote action', () => {
    const note = 'Hi this is a note';
    expect(changeNote(note)).toMatchSnapshot();
  });

  it('should create changeResponse action', () => {
    const response = 'This is a response';
    const id = 2;
    expect(changeResponse(response, id)).toMatchSnapshot();
  });
});
