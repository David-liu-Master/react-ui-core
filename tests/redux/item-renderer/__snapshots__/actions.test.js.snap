// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemRenderer actions should create changeNote action 1`] = `
Object {
  "payload": Object {
    "note": "Hi this is a note",
  },
  "type": "openpatch.ui-core.CHANGE_NOTE",
}
`;

exports[`ItemRenderer actions should create changeResponse action 1`] = `
Object {
  "payload": Object {
    "id": 2,
    "response": "This is a response",
  },
  "type": "openpatch.ui-core.CHANGE_NOTE",
}
`;
